app_flutter/lib/commit_box.dart

```diff
@@ -45,67 +45,70 @@ class _CommitBoxState extends State<CommitBox> {
   }
 
   void _handleTap() {
-    _commitOverlay = this._createCommitOverlay(widget);
+    _commitOverlay = CommitOverlay(widget, context);
     Overlay.of(context).insert(this._commitOverlay);
   }
+}
 
-  OverlayEntry _createCommitOverlay(CommitBox widget) {
-    RenderBox renderBox = context.findRenderObject();
+class CommitOverlay extends OverlayEntry {
+  CommitOverlay(CommitBox widget, BuildContext parentContext)
+      : super(builder: (context) {
+          RenderBox renderBox = parentContext.findRenderObject();
 
-    return OverlayEntry(
-        builder: (context) => Stack(
-              children: <Widget>[
-                // This is the area a user can click (the rest of the screen) to close the overlay.
-                GestureDetector(
-                  onTap: _commitOverlay.remove,
-                  child: Container(
-                    width: MediaQuery.of(context).size.width,
-                    height: MediaQuery.of(context).size.height,
-                    // Color must be defined otherwise the container can't be clicked on
-                    color: Colors.transparent,
-                  ),
+          return Stack(
+            children: <Widget>[
+              // This is the area a user can click (the rest of the screen) to close the overlay.
+              GestureDetector(
+                onTap:
```

Comment:
@jonahwilliams I'm working on refactoring the overlay into its own widget. Everything is working except being able to close out of the widget by clicking outside of it.

All the widget needs is to call remove, but it can't do that since its building that code in the constructor. Do you know of a better method of moving this overlay into its own widget so it can call remove?